examples/ethernet/ethernet/main/ethernet_main.c

```diff
@@ -32,92 +32,45 @@
 #include "tcpip_adapter.h"
 #include "nvs_flash.h"
 #include "driver/gpio.h"
-#include "tlk110_phy.h"
 
-static const char *TAG = "eth_demo";
+extern void phy_device_check_phy_init(void);
```

Comment:
This is really a minor variation, but what about:
```
eth_config_t config = tlk110_default_ethernet_phy_config;
config.gpio_config = eth_gpio_config_rmii;
config.tcpip_input = tcpip_adapter_eth_input;
config.phy_power_enable = phy_device_power_enable;
ret = esp_eth_init(&config);
```

This means that you don't need to selectively compile each PHY source file, they can all be compiled but only one will be picked up at link time (because of the individual name `tlk110_default_ethernet_phy_config`). This will probably help once the code is in a comonent support library (which I agree should be the goal), as many non-example users probably won't need to support multiple PHYs and therefore won't find the overhead of sdkconfig phy selection useful.

Eventually, we can probably also push some of these generic functions (like tcpip_adapter_eth_input) into the "default" config for each PHY as well.